diff --git a/mingwex/stdio/pformat.c b/mingwex/stdio/pformat.c
index 854ad5b..bca6843 100644
--- a/mingwex/stdio/pformat.c
+++ b/mingwex/stdio/pformat.c
@@ -2229,20 +2229,9 @@ int __pformat( int flags, void *dest, int max, const char *fmt, va_list argv )
 	      /* Modifier is `l'; data type is `long' sized...
 	       */
 	      length = PFORMAT_LENGTH_LONG;
-
-#           ifndef _WIN32
-	      /*
-	       * Microsoft's MSVCRT implementation also uses `l'
-	       * as a modifier for `long double'; if we don't want
-	       * to support that, we end this case here...
-	       */
 	      state = PFORMAT_END;
 	      break;
 
-	      /* otherwise, we simply fall through...
-	       */
-#	    endif
-
 	  case 'L':
 	    /*
 	     * Identify the appropriate argument as a `long double',
diff --git a/tlssup.c b/tlssup.c
index e1685f5..3c82448 100644
--- a/tlssup.c
+++ b/tlssup.c
@@ -47,12 +47,12 @@ _CRTALLOC(".CRT$XLZ") PIMAGE_TLS_CALLBACK __xl_z = 0;
 
 #ifdef _WIN64
 _CRTALLOC(".tls") const IMAGE_TLS_DIRECTORY64 _tls_used = {
-  (ULONGLONG) &_tls_start+1, (ULONGLONG) &_tls_end, (ULONGLONG) &_tls_index,
+  (ULONGLONG) &_tls_start, (ULONGLONG) &_tls_end, (ULONGLONG) &_tls_index,
   (ULONGLONG) (&__xl_a+1), (ULONG) 0, (ULONG) 0
 };
 #else
 _CRTALLOC(".tls") const IMAGE_TLS_DIRECTORY _tls_used = {
-  (ULONG)(ULONG_PTR) &_tls_start+1, (ULONG)(ULONG_PTR) &_tls_end,
+  (ULONG)(ULONG_PTR) &_tls_start, (ULONG)(ULONG_PTR) &_tls_end,
   (ULONG)(ULONG_PTR) &_tls_index, (ULONG)(ULONG_PTR) (&__xl_a+1),
   (ULONG) 0, (ULONG) 0
 };
@@ -96,7 +96,8 @@ __dyn_tls_init (HANDLE hDllHandle, DWORD dwReason, LPVOID lpreserved)
       return TRUE;
     }
 
-  for (pfunc = &__xd_a + 1; pfunc != &__xd_z; ++pfunc)
+  /* GCC is optmizing out the intial test causing errors. Removing the +1 fixes */  
+  for (pfunc = &__xd_a; pfunc != &__xd_z; ++pfunc)
     {
       if (*pfunc != NULL)
         (*pfunc)();
