From 3dace856e8d400cdd18fec335a1f4540f3af92ca Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Sat, 22 Jun 2013 06:45:55 -0500
Subject: [PATCH 11/11] Add WinSock2 dependency to phobos

---
 gcc/d/d-spec.c | 36 +++++++++++++++++++++++++++++++++++-
 1 file changed, 35 insertions(+), 1 deletion(-)

diff --git a/gcc/d/d-spec.c b/gcc/d/d-spec.c
index 18f0d18..f19d960 100644
--- a/gcc/d/d-spec.c
+++ b/gcc/d/d-spec.c
@@ -40,6 +40,8 @@
 #define WITHLIBC	(1<<6)
 /* This bit is set when the argument should not be passed to gcc or the backend */
 #define SKIPOPT		(1<<8)
+/* This bit is set if they did `-lws2_32'.  */
+#define WITHLIBWS2_32      (1<<9)
 
 #ifndef MATH_LIBRARY
 #define MATH_LIBRARY "m"
@@ -63,6 +65,14 @@
 #define LIBSTDCXX_PROFILE LIBSTDCXX
 #endif
 
+#ifndef LIBWS2_32
+#define LIBWS2_32 "ws2_32"
+#endif
+#ifndef LIBWS2_32_PROFILE
+#define LIBWS2_32_PROFILE LIBWS2_32
+#endif
+
+
 #ifndef LIBPHOBOS
 #define LIBPHOBOS "gphobos"
 #endif
@@ -70,7 +80,6 @@
 #define LIBPHOBOS_PROFILE LIBPHOBOS
 #endif
 
-
 void
 lang_specific_driver (cl_decoded_option **in_decoded_options,
 		      unsigned int *in_decoded_options_count,
@@ -111,6 +120,9 @@ lang_specific_driver (cl_decoded_option **in_decoded_options,
 
   /* "-lrt" if it appears on the command line.  */
   const cl_decoded_option *saw_time = 0;
+  
+  /* "-lws2_32" if it appears on the command line.  */
+  const cl_decoded_option *saw_libws2_32 = 0;  
 
   /* "-lc" if it appears on the command line.  */
   const cl_decoded_option *saw_libc = 0;
@@ -245,6 +257,8 @@ lang_specific_driver (cl_decoded_option **in_decoded_options,
 	    }
 	  else if (strcmp (arg, "c") == 0)
 	    args[i] |= WITHLIBC;
+      else if (strcmp (arg, "ws2_32") == 0)
+        args[i] |= WITHLIBWS2_32;		
 	  else
 	    /* Unrecognized libraries (e.g. -ltango) may require libphobos.  */
 	    library = (library == 0) ? 1 : library;
@@ -412,6 +426,12 @@ lang_specific_driver (cl_decoded_option **in_decoded_options,
 	  --j;
 	  saw_libc = &decoded_options[i];
 	}
+	
+      if (!saw_libws2_32 && (args[i] & WITHLIBWS2_32) && library > 0)
+    {
+      --j;
+      saw_libws2_32 = &decoded_options[i];
+    }	
 
       if (!saw_libcxx && (args[i] & WITHLIBCXX) && library > 0)
 	{
@@ -547,6 +578,9 @@ lang_specific_driver (cl_decoded_option **in_decoded_options,
 
   if (saw_libc)
     new_decoded_options[j++] = *saw_libc;
+  
+  if (saw_libws2_32)
+    new_decoded_options[j++] = *saw_libws2_32;
 
   if (shared_libgcc && !static_link)
     {
-- 
1.8.5.2.msysgit.0

