From 61e3e3c879eb8b3ce279b0701db194eaa319de56 Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Sat, 22 Jun 2013 07:05:44 -0500
Subject: [PATCH 11/18] DLLIMPORT bugfix. TLS bugfix

---
 gcc/d/d-objfile.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/d/d-objfile.cc b/gcc/d/d-objfile.cc
index 733a5d2..73eb47e 100644
--- a/gcc/d/d-objfile.cc
+++ b/gcc/d/d-objfile.cc
@@ -1260,7 +1260,7 @@ Module::genobjfile (int)
 
 #ifdef TARGET_DLLIMPORT_DECL_ATTRIBUTES
     // Apply dllexport for ModuleInfo and dllimport for imports.
-    gen.addDeclAttribute(toSymbol()->Stree, "dllexport");
+    insert_decl_attribute (toSymbol()->Stree, "dllexport");
 
       for (size_t i = 0; i < aimports.dim; i++)
     {   
@@ -1269,7 +1269,7 @@ Module::genobjfile (int)
         if (m->needmoduleinfo)
       {   
         Symbol *s = m->toSymbol();                
-        gen.addDeclAttribute(s->Stree, "dllimport");
+        insert_decl_attribute (s->Stree, "dllimport");
         DECL_DLLIMPORT_P(s->Stree) = 1;
       }
     }
-- 
1.8.1.msysgit.1

