From c67edef06cfe75ffc71edccb6d0e1b3a3c04081d Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Sat, 22 Jun 2013 07:16:41 -0500
Subject: [PATCH] Make ClassInfo DLL_IMPORT & DLL_EXPORT

---
 gcc/d/d-decls.cc | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/gcc/d/d-decls.cc b/gcc/d/d-decls.cc
index 052dccf..d45a14d 100644
--- a/gcc/d/d-decls.cc
+++ b/gcc/d/d-decls.cc
@@ -586,6 +586,17 @@ Module::toSymbol (void)
 	}
 
       csym->ScontextDecl = module;
+
+#if TARGET_WINDOS && TARGET_DLLIMPORT_DECL_ATTRIBUTES
+      // Have to test for import first
+      if (isImportedSymbol())
+        {
+          gen.addDeclAttribute(decl, "dllimport");
+          DECL_DLLIMPORT_P(decl) = 1;
+        }
+      else if (isExport())
+        gen.addDeclAttribute(decl, "dllexport");
+#endif  	  
     }
 
   return csym;
-- 
1.8.5.2.msysgit.0

