From 3280e19b899f81186070cc324c867afbe482b14e Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Sat, 22 Jun 2013 07:20:15 -0500
Subject: [PATCH 15/16] Merge with other TLS commits

---
 gcc/d/d-objfile.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gcc/d/d-objfile.cc b/gcc/d/d-objfile.cc
index 73eb47e..e649508 100644
--- a/gcc/d/d-objfile.cc
+++ b/gcc/d/d-objfile.cc
@@ -1673,12 +1673,14 @@ d_finish_function (FuncDeclaration *f)
 
   gcc_assert (TREE_CODE (t) == FUNCTION_DECL);
 
-  // Write out _tlsstart/_tlsend.
   // For MinGW we alias _tls_start and _tls_end.  Otherwise we have an offset
   // that could result in memory corruption.  As well as alignment issues for
   // the GC.
-  if (!TARGET_WINDOS && (f->isMain() || f->isWinMain() || f->isDllMain()))
+#ifndef TARGET_WINDOS  
+  // Write out _tlsstart/_tlsend.
+  if (f->isMain() || f->isWinMain() || f->isDllMain())
     build_tlssections();
+#endif
 
   if (s->prettyIdent)
     DECL_NAME (t) = get_identifier (s->prettyIdent);
-- 
1.8.1.msysgit.1

