From 210059211a38ab4a732b5300a9976aa7c2e17041 Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Tue, 10 Sep 2013 21:40:57 -0500
Subject: [PATCH 18/18] insert_decl_attributes renamed insert_decl_attribute

---
 gcc/d/d-decls.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gcc/d/d-decls.cc b/gcc/d/d-decls.cc
index 8e2e6a5..dead8e0 100644
--- a/gcc/d/d-decls.cc
+++ b/gcc/d/d-decls.cc
@@ -209,11 +209,11 @@ VarDeclaration::toSymbol (void)
       // Have to test for import first
       if (isImportedSymbol())
 	{
-	  insert_decl_attributes (var_decl, "dllimport");
+	  insert_decl_attribute (var_decl, "dllimport");
 	  DECL_DLLIMPORT_P (var_decl) = 1;
 	}
       else if (isExport())
-	insert_decl_attributes (var_decl, "dllexport");
+	insert_decl_attribute (var_decl, "dllexport");
 #endif
 
       if (global.params.vtls && isDataseg() && isThreadlocal())
@@ -449,11 +449,11 @@ FuncDeclaration::toSymbol (void)
 	  // Have to test for import first
 	  if (isImportedSymbol())
 	    {
-	      insert_decl_attributes (fndecl, "dllimport");
+	      insert_decl_attribute (fndecl, "dllimport");
 	      DECL_DLLIMPORT_P (fndecl) = 1;
 	    }
 	  else if (isExport())
-	    insert_decl_attributes (fndecl, "dllexport");
+	    insert_decl_attribute (fndecl, "dllexport");
 #endif
 	  set_decl_location (fndecl, this);
 	  setup_symbol_storage (this, fndecl, false);
-- 
1.8.1.msysgit.1

