From 1b2ea560507235c53a48bf6e7e099843e0ad8fe4 Mon Sep 17 00:00:00 2001
From: Daniel Green <venix1@gmail.com>
Date: Thu, 14 Nov 2013 19:25:03 -0600
Subject: [PATCH] MinGW64 corrections.

---
 libphobos/libdruntime/core/sys/windows/mingwasm.S | 36 +++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/libphobos/libdruntime/core/sys/windows/mingwasm.S b/libphobos/libdruntime/core/sys/windows/mingwasm.S
index 49b5918..16cf089 100644
--- a/libphobos/libdruntime/core/sys/windows/mingwasm.S
+++ b/libphobos/libdruntime/core/sys/windows/mingwasm.S
@@ -3,6 +3,39 @@
  */
  	.global	_fiber_switchContext
 _fiber_switchContext:
+#if defined(__x86_64__)
+	// save current stack state
+    pushq %RBP;
+    movq  %RSP, %RBP;
+    pushq %RBX;
+    pushq %R12;
+    pushq %R13;
+    pushq %R14;
+    pushq %R15;
+    pushq %GS:0;
+    pushq %GS:8;
+    pushq %GS:16;
+	
+	// store oldp
+    movq %RSP, (%RCX);
+    // load newp to begin context switch
+    movq %RDX, %RSP;
+
+    // load saved state from new stack
+    popq %GS:16;
+	popq %GS:8;
+    popq %GS:0;
+    popq %R15;
+    popq %R14;
+    popq %R13;
+    popq %R12;
+    popq %RBX;
+    popq %RBP;
+
+	// 'return' to complete switch
+    popq %RCX;
+    jmp *%RCX;
+#elif defined(__x86__)
 	// Save current stack state.save current stack state
 	// Standard CDECL prologue.  
 	push %EBP;
@@ -33,3 +66,6 @@ _fiber_switchContext:
 
     // 'return' to complete switch
     ret;
+#else
+// assert?
+#endif
\ No newline at end of file
-- 
1.9.2.msysgit.0

